channels/routing.py

```diff
@@ -52,6 +54,71 @@ async def __call__(self, scope, receive, send):
             )
 
 
+def _parse_resolver(child_url_pattern, parent_resolver, parent_regex, routes):
+    """Parse resolver (returned by `include`) recurrsively
+
+    Parameters
+    ----------
+    child_url_pattern : URLResolver |
+        The child url pattern
+    parent_resolver : URLResolver
+        The parent resolver
+    parent_regex : Pattern
+        The parent regex pattern
+    routes : list[URLPattern]
+        The URLPattern's list that stores the routes
+
+    Returns
+    -------
+    list[URLPattern]
+        The URLPattern's list that stores the routes
+    """
+    if not child_url_pattern.callback and isinstance(child_url_pattern, URLResolver):
```

Comment:
Should we change this line to:

```
if isinstance(child_url_pattern, URLResolver) and len(child_url_pattern.url_patterns) > 0:
```

It seems more clear to me, but maybe there's a problem with my suggestion that I'm not seeing?

Reply:
Actually, I imitated the original channels code

https://github.com/django/channels/blob/e5331869fd7b26dd396ea1f60743feadfc1f4ca2/channels/routing.py#L85

So, there is no reason to check `not child_url_pattern.callback` ; )

I think `len(child_url_pattern.url_patterns) > 0` is unnecessary because `url_patterns` of  `URLResolver` is list

Reply:
That's why should I modify this code into `if isinstance(child_url_pattern, URLResolver)` only?

Reply:
I removed `not child_url_pattern.callback` for now by https://github.com/django/channels/pull/2110/commits/dd4b1aa822eed6816151641a3f4410c3141e8e77